### DISCUSSION TITLE
Reconsider our world coordinate system

### DISCUSSION THREAD

**Discussion by philpax** [AmbientRun/Ambient#547] [Ideas]
Our world coordinate system, as described [here](https://ambientrun.github.io/Ambient/reference/runtime.html), is left-handed +x right +y back +z up, with the reasoning that this is similar to Unreal. However:

- Unreal is alone in this particular convention of left-handed z-up: https://twitter.com/freyaholmer/status/1325556229410861056
- Unreal's coordinate system, as far as I can tell, is +x forward, +y right, +z up, which means our scheme is uniquely odd

This leads to a lot of confusion when handling assets from other platforms, as well as generally doing transformations; we don't appear to share a coordinate scheme with *any* other engines or applications.

Should we change this while we still have the chance?

**Comment by ten3roberts** [comment]
Another thing to note is that `glam` assumes and suggests Y up left-hand coordinates for euler angles and conversions. These docs are visible and accessible inside our scripting, but breaks the assumption when used
